src/components/Navbar/Drawer/index.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useRef, useEffect, useCallback, ReactNode } from 'react';
+
+import classNames from 'classnames';
+import { useRouter } from 'next/router';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+
+import styles from './Drawer.module.scss';
+import DrawerCloseButton from './DrawerCloseButton';
+
+import useKeyPressedDetector from 'src/hooks/useKeyPressedDetector';
+import useOutsideClickDetector from 'src/hooks/useOutsideClickDetector';
+import {
+  Navbar,
+  selectNavbar,
+  setIsNavigationDrawerOpen,
+  setIsSearchDrawerOpen,
+  setIsSettingsDrawerOpen,
+} from 'src/redux/slices/navbar';
+
+export enum DrawerType {
+  Navigation = 'navigation',
+  Search = 'search',
+  Settings = 'settings',
+}
+
+export enum DrawerSide {
+  Left = 'left',
+  Right = 'right',
+}
+
+interface Props {
+  type: DrawerType;
+  side?: DrawerSide;
+  header: ReactNode;
+  children: ReactNode;
+}
+
+/**
+ * Check whether a specific drawer is open or not based on the type.
+ *
+ * @param {DrawerType} type
+ * @param {Navbar} navbar
+ * @returns {boolean}
+ */
+const getIsOpen = (type: DrawerType, navbar: Navbar): boolean => {
+  const { isNavigationDrawerOpen, isSettingsDrawerOpen, isSearchDrawerOpen } = navbar;
+  if (type === DrawerType.Navigation) {
+    return isNavigationDrawerOpen;
+  }
+  if (type === DrawerType.Settings) {
+    return isSettingsDrawerOpen;
+  }
+  return isSearchDrawerOpen;
+};
+
+const getActionCreator = (type: DrawerType) => {
+  if (type === DrawerType.Navigation) {
+    return setIsNavigationDrawerOpen.type;
+  }
+  if (type === DrawerType.Settings) {
+    return setIsSettingsDrawerOpen.type;
+  }
+  return setIsSearchDrawerOpen.type;
+};
```

Comment:
quick question 

is it possible to make the Drawer component not aware of the `type`. So it can be more "polymorphism" / decoupled from detail